malaria-app-android/src/main/java/com/peacecorps/malaria/activities/RapidFireGame.java

```diff
@@ -236,17 +241,31 @@ public RapidFireTimeCounter(long millisInFuture, long countDownInterval) {
         @Override
         public void onTick(long l) {
             timer.setText(""+l/1000);
+            millisLeft=l;
         }
 
         @Override
         public void onFinish() {
             counter.cancel();
             prepNextQues();
         }
+    }
 
-
+    @Override
+    protected void onPause() {
+        counter.cancel();
+        counter= null;
+        super.onPause();
     }
 
+    @Override
+    protected void onResume() {
+        if(counter==null){
+            counter= new RapidFireTimeCounter(millisLeft,1000);
```

Comment:
Creating a new timer every time is not the best way to go. Utilize .start and .cancel in onPause and onResume and work with a single timer. Also I dont believe there is a need to set the timer to null while cancelling it.